### PR TITLE
Update/auto redirect after install condition

### PR DESCRIPTION
**Pull request by jboland88**
Adds a method to detect plugin install context so that we can conditionally handle redirection after plugins are activated.

#### Changes proposed in this Pull Request:

* Adds a new function in `Plugins_Installer` to check the plugin installation context
* Adds a conditional check in the protect plugin that calls the new function before redirecting on plugin activation.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

No

##### Reproduce the issue
* To see the original issue, you can test locally with the trunk branch checked out.
* First, try to enable the protect plugin from the CLI: `wp plugin activate protect`. If you are testing with the docker environment, you can run this using `jetpack docker wp plugin activate protect`.
* You will get a `Warning: Some code is trying to do a URL redirect.` message with an error backtrace. The protect plugin is still enabled, but the return is not clean.
* To test activating a plugin over the API, we can temporarily change the Creative Mail recommendation logic used to install the Creative Mail plugin in the recommendations flow. Update [this line](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/client/recommendations/feature-utils.js#L96) in `feature-utils.js` so that the plugin installed is `protect`.
* You will need to re-build the Jetpack plugin with `jetpack build plugins/jetpack`.
* You will also need to deactivate the protect plugin before the next step.
* Now, visit `/wp-admin/admin.php?page=jetpack#/recommendations/summary` and make sure your browser inspector console is open. Click the "Install" button next to the "Creative Mail" recommendation. You should get a `JsonParseAfterRedirectError` console error.

##### Test the change:
* Stash the change that was made to the `feature-utils.js` file in the previous steps
* Checkout the feature branch, apply the change from the stash and re-build the Jetpack plugin
* Now, making sure the protect plugin is not currently active, repeat the CLI testing steps from above. The return should be clean and the plugin should be activated
* Deactivate the protect plugin
* Now, repeat the steps above to activate the protect plugin from the recommendations summary screen. This time, you should not get any console errors or messages.
* Visit the plugins page, deactivate the protect plugin
* Now, from the plugins page, activate the protect plugin and confirm that you are redirected to the protect plugin page after activation - this is expected only when activating from the plugins screen.
* Test activating the protect plugin from the plugins screen as part of a bulk activation. You should NOT be redirected in this case.

### CODE REVIEW

**Comment by CodeyGuyDylan** [comment]
I fixed a compatibility issue and it looks like some of the versions are out of whack (last time this happened to me they said some of my code was caught in between a version release) but other than that this seems to work well

File: projects/packages/plugins-installer/src/class-plugins-installer.php [hunk 1]
```diff
@@ -235,4 +235,53 @@ public static function get_plugins() {
...
 		return array();
 	}
+
+	/**
+	 * Determine if the current request is activating a plugin from the plugins page.
+	 *
+	 * @param string $plugin Plugin file path to check.
+	 * @return bool
+	 */
+	public static function is_current_request_activating_plugin_from_plugins_screen( $plugin ) {
```
Comment (leogermani) [inline projects/packages/plugins-installer/src/class-plugins-installer.php:245]: I like the verbose name! :)

File: projects/plugins/protect/jetpack-protect.php [hunk 2]
```diff
@@ -97,7 +97,10 @@ function () {
  * @param string $plugin Path to the plugin file relative to the plugins directory.
  */
 function jetpack_protect_plugin_activation( $plugin ) {
-	if ( JETPACK_PROTECT_ROOT_FILE_RELATIVE_PATH === $plugin ) {
+	if (
```
Comment (leogermani) [inline projects/plugins/protect/jetpack-protect.php:100]: nitpick, I think we could invert the order of the checks, as `JETPACK_PROTECT_ROOT_FILE_RELATIVE_PATH === $plugin` is a much cheaper check.

Comment (jboland88) [reply projects/plugins/protect/jetpack-protect.php:100]: Good call, thanks!

File: projects/packages/plugins-installer/src/class-plugins-installer.php (hunk 1 above)
Comment (jboland88) [reply projects/packages/plugins-installer/src/class-plugins-installer.php:245]: Thanks!

**Review by elliottprogrammer — DISMISSED** [review]
Fully tested this per instructions and it all works good, exactly as described. The code looks all good to me.
LGTM! 👍

**Review by leogermani — DISMISSED** [review]

**Review by kraftbj — DISMISSED** [review]
Re-approving based on @leogermani's earlier approval that was cycled due to a lock file change.

**Review by kraftbj — APPROVED** [review]
Needed to re-run the lock after I updated things locally, putting me at PHP 8.1 without me realizing it.

**Pull request merged**